keyboards/redox/keymaps/jeherve/keymap.c

```diff
@@ -12,126 +29,127 @@ extern keymap_config_t keymap_config;
-			}
-			return false;
-			break;
-    case INV_1P:
-      if (record->event.pressed) {
-        SEND_STRING(SS_DOWN(X_LGUI)SS_DOWN(X_LALT)SS_TAP(X_BSLASH)SS_UP(X_LGUI)SS_UP(X_LALT));
-      }
-      return false;
-      break;
-    case ALFRED:
-      if (record->event.pressed) {
-        SEND_STRING(SS_DOWN(X_LALT)SS_TAP(X_SPACE)SS_UP(X_LALT));
-      }
-      return false;
-      break;
-		case YOSHI:
-			if (record->event.pressed) {
-				SEND_STRING(":yellow_yoshi:");
-			}
-			return false;
-			break;
-		case THUMB_UP:
-			if (record->event.pressed) {
-				SEND_STRING(SS_LALT("D83D+DC4D"));
-			}
-			return false;
-			break;
-		case WAVE:
-			if (record->event.pressed) {
-				SEND_STRING(SS_LALT("D83D+DC4B"));
-			}
-			return false;
-			break;
-		case NBSP:
-			if (record->event.pressed) {
-				SEND_STRING("&nbsp;");
-			}
-			return false;
-			break;
-	case KC_LSFT:
-		shift_held = record->event.pressed;
-		return true;
-		break;
-	case KC_RSFT:
-		shift_held = record->event.pressed;
-		return true;
-		break;
-		case M_BRACKET_LEFT: {
-			if (record->event.pressed) {
-				if (shift_held) {
-					unregister_code(KC_LSFT);
-					unregister_code(KC_RSFT);
-					register_code(KC_LBRC);
-				} else {
-					register_code(KC_LSFT);
-					register_code(KC_9);
-				}
-			} else { // Release the key
-				unregister_code(KC_LBRC);
-				unregister_code(KC_LSFT);
-				unregister_code(KC_RSFT);
-				unregister_code(KC_9);
-			}
-			return false;
-			break;
-		}
-		case M_BRACKET_RIGHT: {
-			if (record->event.pressed) {
-				if (shift_held) {
-					unregister_code(KC_LSFT);
-					unregister_code(KC_RSFT);
-					register_code(KC_RBRC);
-				} else {
-					register_code(KC_LSFT);
-					register_code(KC_0);
-				}
-			} else { // Release the key
-				unregister_code(KC_RBRC);
-				unregister_code(KC_LSFT);
-				unregister_code(KC_RSFT);
-				unregister_code(KC_0);
-			}
-			return false;
-			break;
-		}
-	}
-	return true;
+    switch (keycode) {
+        case SHRUG:
+            if (record->event.pressed) {
+                send_unicode_string("¯\\_(ツ)_/¯");
+            }
+            return false;
+            break;
+        case INV_1P:
+            if (record->event.pressed) {
+                SEND_STRING(SS_DOWN(X_LGUI) SS_DOWN(X_LALT) SS_TAP(X_BSLASH) SS_UP(X_LGUI) SS_UP(X_LALT));
+            }
+            return false;
+            break;
+        case ALFRED:
+            if (record->event.pressed) {
+                SEND_STRING(SS_DOWN(X_LALT) SS_TAP(X_SPACE) SS_UP(X_LALT));
+            }
+            return false;
+            break;
+        case YOSHI:
+            if (record->event.pressed) {
+#ifdef AUDIO_ENABLE
+                PLAY_SONG(tone_zelda);
+#endif
+                SEND_STRING(":yellow_yoshi:");
+            }
+            return false;
+            break;
+        case THUMB_UP:
+            if (record->event.pressed) {
+                SEND_STRING(SS_LALT("D83D+DC4D"));
```

Comment:
Is this the correct way to input surrogate pairs on macOS? I see there is some special case in `register_unicode()` for it that simply extracts the pair from the codepoint and sends them one after the other - no `+` in between. Maybe that's a bug.

Reply:
Would you like to test without the `+`, see how that goes, or would I need to build from the branch in #12751?

I believe I took inspiration from the implementation in [this post](https://sevenseacat.net/posts/2018/unicode-in-qmk-on-osx/) when I first built that keymap, and it's been working well ever since, but I'm happy to try something new, if that can help.

Reply:
Try without the +, and you can simply patch in that one line from my PR temporarily and replace your sendstring with `register_unicode(0x1F44D)` instead. I'm interested to know which of those two work.

Reply:
As it turns out, this is wrong on two counts - the keycodes apparently cannot be shifted, and it does not work with the `+` either. So my PR is unnecessary, and you can simply do:
```suggestion
                register_unicode(0x1F44D); // 👍
```

Reply:
That worked well, thank you! 👍 I committed the changes in d9b7d3bf5536376945c5a6348b49cb309d1afa93